### PR TITLE
draw.aaline: blend correctly

### PR DESCRIPTION
**Pull request by dlon**
#575 

Difference:

### CODE REVIEW

File: src_c/draw.c [hunk 1]
```diff
@@ -102,10 +102,6 @@ aaline(PyObject *self, PyObject *arg)
         return NULL;
     surf = pgSurface_AsSurface(surfobj);
 
-    if (surf->format->BytesPerPixel != 3 && surf->format->BytesPerPixel != 4)
```
Comment (illume) [inline src_c/draw.c]: oh cool. I guess this means it supports more than 32/24 bit now.

File: src_c/draw.c [hunk 2]
```diff
@@ -983,27 +975,75 @@ set_at(SDL_Surface *surf, int x, int y, Uint32 color)
...
+        case 1:
+            return *pixels;
+    }
+    return 0;
+}
+
+static void
+set_pixel_32(Uint8 *pixels, SDL_PixelFormat *format, Uint32 pixel)
+{
+    switch (format->BytesPerPixel) {
```
Comment (e1000) [inline src_c/draw.c]: Cool! It looks similar to the code in **drawline** now, but why it is much more simple here? Can we use this code in *drawline* ?

File: src_c/draw.c [hunk 3]
```diff
@@ -983,27 +975,75 @@ set_at(SDL_Surface *surf, int x, int y, Uint32 color)
...
+    if (blend) \
+        draw_pixel_blended_32(pixels, colorptr, br, surf->format); \
+    else {\
+        set_pixel_32(pixels, surf->format, \
+                     SDL_MapRGBA(surf->format, \
+                                 br * colorptr[0], \
+                                 br * colorptr[1], \
+                                 br * colorptr[2], \
+                                 br * colorptr[3])); \
+    }}
```
Comment (e1000) [inline src_c/draw.c]: maybe @illume would like to apply some *clang-format* ? ;)

**Comment by dlon** [comment]
Some measurements:

Not inlined:
  18985 calls/s
Inlined:
  18854 calls/s
Replaced trunc
  23240 calls/s
Replaced trunc + inlined:
  22826 calls/s

Inlining seems to make little difference when I try it, but replacing trunc() made a big difference.

Also, the code for big endian might be wrong.

File: src_c/draw.c (hunk 1 above)
Comment (dlon) [reply src_c/draw.c]: I think it's always between 1 and 4, but 2 and 1 also work now

File: src_c/draw.c (hunk 2 above)
Comment (dlon) [reply src_c/draw.c]: In drawline, the line is drawn for each case inside the switch statement, probably because it might be faster.

**Comment by e1000** [comment]
* `x` is for expected Failure,
* `u` is for unexpected Success,

so it seems, no unexpected success :-/ I will investigate that "ASAP"...

**Pull request merged**